tests/unit_tests/test_dataset.py

```
@@ -95,3 +95,95 @@ def test_find_best_score(simple_df):
     assert score == "x"
     assert npass == 4
     assert desc
+
+
+# Validation and error handling ---------------------------------------------------
+def test_no_decoys_raises(simple_df):
+    """PsmDataset with only target PSMs must raise ValueError."""
+    targets_only = simple_df[simple_df["target"] == True].copy()
```

Ruff (0.15.17)

[error] Avoid equality comparisons to `True`; use `simple_df["target"]:` for truth checks

Replace with `simple_df["target"]`

(E712)